### DISCUSSION TITLE
Workflow: Set a Specific Key Value within a property.

### DISCUSSION THREAD

**Discussion by JStafford00** [OrchardCMS/OrchardCore#11697] [Q&A]
So, within a workflow, I have a property set that holds an array of json objects. I'm currently running through a  foreach loop and accessing each object and comparing a certain value. If the value isn't true, I do some code and then set the value to true and restore it into the property field.

To access and set the value, I'm trying to do this within a script task, but it seems like you can't set a key value within a json in a script task (At least I can't the way I'm doing it.)

![image](https://user-images.githubusercontent.com/42542054/168379120-89c3d6e2-c1cb-48de-92a9-dafcd761b8f3.png)

The array looks a bit like this, and can be any number of json objects in it.

![image](https://user-images.githubusercontent.com/42542054/168378052-74c0a7db-a323-4ffd-8a7b-2d049264f6d2.png)

The output I am getting, is that the key value is not being set when I try to set it. I made sure by setting two other temporary properties with the before and after of trying to set the value and emailing the json to myself.

Edit: Ignore the "==", that was a typo.

Edit2: Fixed the typos on the images.

**Comment by lampersky** [comment]
if you want to read a property value, please use `var temp = property("variableName");`
to change property value use `setProperty("variableName", 12345);`

here you can find more examples:
https://github.com/lampersky/OrchardCoreWorkflowsExamples
and here you can find other useful methods
https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Workflows/Scripting/WorkflowMethodsProvider.cs

**Reply by JStafford00** [reply]
My problem is I can't set "output.foo.Value = true". I can set the whole property just fine. Index property increments after script is ran, but the output property stays the same. I found the problem is when I try to set the "Value" key to true. This doesn't actually update it in the local output variable.

**Reply by lampersky** [reply]
if you want to change complex object like array, first you have to read it, modify and save again:
```
var array = property('my-array');
//let's say its value is [{ 'key1': 'value2' }, { 'key2': 'value3' }, { 'key3': 'value4' }, { 'key4': 'value5' }]
array['key2'] = 'new value';
setProperty('my-array', array);
```

**Reply by lampersky** [reply]
another solution is to store all items as properties:
```
var myArray = [{ 'foo': 'value2' }, { 'foo': 'value3' }, { 'foo': 'value4' }, { 'foo': 'value5' }];
var idx = 0;
for (const item of myArray) {
    setProperty(`item-${idx++}`, item['foo']);
}
setProperty('count', myArray.length);
```
and this way, you will be able to access particular item with property() and you can change its value with setProperty()

```
var temp = property('item-0');
setProperty('item-0', 'new value');
```
